OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -98,23 +101,22 @@ public void Flash()
 
 		public IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
-			if (needRenderables)
+			if (needRenderables && (renderables == NoRenderables || DamageStateCached != DamageState || OwnerCached != Owner))
```

Comment:
These extra checks don't look right to me.  if needRenderables is true, then I would expect it to generate renderables.  If there is some condition where you don't want it to generate renderables, then those conditions should be applied at the point that variable is set.